addons/purchase_mrp/report/mrp_report_bom_structure.py

```diff
@@ -12,6 +13,9 @@ def _format_route_info(self, rules, rules_delay, warehouse, product, bom, quanti
         if self._is_buy_route(rules, product, bom):
             buy_rules = [rule for rule in rules if rule.action == 'buy']
             supplier = product._select_seller(quantity=quantity, uom_id=product.uom_id)
```

Comment:
some sort of uom conversion is needed here, but which :man_shrugging:

easy way to check:
- component seller qty/uom = 24 units
- bom line qty/uom = 2 dozen

Currently in stable it doesn't find this seller. This is only a semi-related bug, not caused by your changes